File: app/src/main/java/com/ashu/callapitestcode/data/adapter/ForecastAdapter.java
package com.ashu.callapitestcode.data.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.ashu.callapitestcode.R;
import com.ashu.callapitestcode.data.model.ForecastItem;
import com.ashu.callapitestcode.other.graphs.ImageLoaderUtil;

import java.util.List;

public class ForecastAdapter extends RecyclerView.Adapter<ForecastAdapter.ViewHolder> {

    private List<ForecastItem> list;
    private Context context;

    public ForecastAdapter(Context context, List<ForecastItem> list) {
        this.context = context;
        this.list = list;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_forecast, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        ForecastItem item = list.get(position);

        holder.tvDay.setText(item.getDay());
        holder.tvTemp.setText(item.getMaxTemp() + "°/" + item.getMinTemp() + "°");
        holder.tvRain.setText("☁"+item.getRain() + "%");

        // SVG load (your method)
        ImageLoaderUtil.loadSvgIntoImageView(context, item.getIcon(), holder.imgIcon);
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvDay, tvTemp, tvRain;
        ImageView imgIcon;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvDay = itemView.findViewById(R.id.tvDay);
            tvTemp = itemView.findViewById(R.id.tvTemp);
            tvRain = itemView.findViewById(R.id.tvRain);
            imgIcon = itemView.findViewById(R.id.imgIcon);
        }
    }
}

File: app/src/main/java/com/ashu/callapitestcode/data/adapter/ForecastAdapter.kt
package com.ashu.callapitestcode.data.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.ashu.callapitestcode.R
import com.ashu.callapitestcode.data.model.ForecastItem
import com.ashu.callapitestcode.other.graphs.ImageLoaderUtil

class ForecastAdapter(
    private val context: Context,
    private val list: List<ForecastItem>
) : RecyclerView.Adapter<ForecastAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context)
            .inflate(R.layout.item_forecast, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = list[position]

        holder.tvDay.text = item.day
        holder.tvTemp.text = "${item.maxTemp}°/${item.minTemp}°"
        holder.tvRain.text = "☁${item.rain}%"

        // SVG load
        ImageLoaderUtil.loadSvgIntoImageView(context, item.icon, holder.imgIcon)
    }

    override fun getItemCount(): Int = list.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvDay: TextView = itemView.findViewById(R.id.tvDay)
        val tvTemp: TextView = itemView.findViewById(R.id.tvTemp)
        val tvRain: TextView = itemView.findViewById(R.id.tvRain)
        val imgIcon: ImageView = itemView.findViewById(R.id.imgIcon)
    }
}